R/import_metadata_and_counts.R

```diff
@@ -0,0 +1,118 @@
+  # check count H5 directory name not included in the cache directory original
+  all(!metadata_tbl$file_id_db %in% dir(original_dir)) |>
+    check_true() |>
+    assert_that(msg = "Count H5 directory name should not duplicate with that in cache directory")
+
+  counts_data <- sce_obj@assays@data$X
+  assert_that(inherits(sce_obj, "SingleCellExperiment"),
+              msg = "SingleCellExperiment Object is not created from metadata.")
+  assert_that(!"^ENS" %in% genes,
+              msg = "Gene names cannot contain Ensembl IDs.")
+  assert_that(all(counts_data >= 0),
+              msg = "Counts for SingleCellExperiment cannot be negative.")
+  
+  # check the metadata contains cell_, file_id_db, sample_ with correct types
+  check_true("cell_" %in% names(metadata_tbl))
+  check_true("file_id_db" %in% names(metadata_tbl)) 
+  select(metadata_tbl, .data$cell_) |> class() |> check_character()
+  select(metadata_tbl, .data$file_id_db) |> class() |> check_character()
+  #metadata_tbl |> select(cell_, file_id_db) |> sapply(class) |> check_character()
+  
+  # check cell_ values in metadata_tbl is unique
+  (anyDuplicated(metadata_tbl$cell_) == 0 ) |> assert_that(msg = "cell_ in the metadata must be unique")
+  
+  # check cell_ values are not duplicated when join with parquet
+  cells <- select(get_metadata(), .data$cell_) |> as_tibble()
+  (!any(metadata_tbl$cell_ %in% cells$cell_)) |> assert_that(msg = "cell_ in the metadata should not duplicate with that exists in API")
+  
+  # check age_days is either -99 or greater than 365
+  if (any(colnames(metadata_tbl) == "age_days")) {
+    assert_that(all(metadata_tbl$age_days==-99 | metadata_tbl$age_days> 365),
+                msg = "age_days should be either -99 for unknown or greater than 365")
+  }
+  
+  # check sex capitalisation then convert to lower case 
+  if (any(colnames(metadata_tbl) == "sex")) {
+    metadata_tbl <- metadata_tbl |> mutate(sex = tolower(.data$sex))
+    dplyr::distinct(metadata_tbl, .data$sex) |> dplyr::pull() |> check_subset(c("female","male","unknown"))
+  }
+  counts_path <-
+    select(metadata_tbl, .data$file_id_db) |> mutate(
+      original_path = file.path(original_dir, basename(.data$file_id_db)),
+      cpm_path = file.path(cache_dir, "cpm", basename(.data$file_id_db))
+    )
+  
+  # if checkpoints above pass, generate cpm
+  cli_alert_info("Generating cpm from {.path {metadata_tbl$file_id_db}}. ")
+  
+  # Generate cpm from counts
+  get_counts_per_million(input_sce_obj = sce_obj, output_dir = counts_path$cpm_path, hd5_file_dir = counts_path$original_path)
+  saveHDF5SummarizedExperiment(sce_obj, counts_path$original_path, replace=TRUE)
+  
+  cli_alert_info("cpm are generated in {.path {counts_path$cpm_path}}. ")
+  
+  # check metadata sample file ID match the count file ID in cache directory
+  all(metadata_tbl |> pull(.data$file_id_db) %in% dir(original_dir)) |> 
+    assert_that(msg = "The metadata sample file ID and the count file ID does not match")
```

Comment:
It seems that you are checking that the files were correctly created. If so, the error message needs to be updated.